Assets/_Functionalities/ObjImporter/Scripts/ObjSpawner.cs

```diff
@@ -84,7 +102,43 @@ private void OnObjImported(GameObject returnedGameObject)
 
         private void DisposeImporter()
         {
-            if (importer != null) Destroy(importer.gameObject);
+            if (importer != null)
+            {
+                importer.MtlImportSucceeded.RemoveListener(MtlImportSuccess.Invoke);
+                Destroy(importer.gameObject);
+            }
+        }
+
+        public void SetObjPathInPropertyData(string fullPath)
```

Comment:
kunnen de volgende methods niet generic?

Reply:
hoe dan? dit is specifiek voor de ObjPropertyData omdat die een ObjFile property heeft

Reply:
excuus ik bedoelde niet generics maar generiek opzetten

public void SetPathInPropertyData(string fullPath, ref System.Uri uri)
{
    var propertyData = PropertyData as ObjPropertyData;
    uri = AssetUriFactory.CreateProjectAssetUri(fullPath);
}

private string GetPathFromPropertyData(System.Uri uri)
{
    if (uri == null)
        return "";

    var localPath = uri.LocalPath.TrimStart('/', '\\');
    var path = Path.Combine(Application.persistentDataPath, localPath);
    return path;
}

Reply:
hm, ik ben geen fan van ref argumenten gebruiken tenzij het echt moet, en volgens mij set die functie nu niet de uri in PropertyData.ObjFile